### PR TITLE
Feature/a b graph [ PART 1 ]

### PR DESCRIPTION
**Pull request by pNagi**
- SessionType selector
- query nodes links and path filter by version (featureId and name)

### CODE REVIEW

File: src/server/graphql/QueryRootType.js [hunk 1]
```diff
@@ -1,19 +1,23 @@
 import {
   GraphQLList,
+  GraphQLNonNull,
   GraphQLObjectType,
   GraphQLString,
 } from 'graphql'
+import _ from 'lodash'
```
Comment (0xchimz) [inline src/server/graphql/QueryRootType.js:7]: `_` is never used

File: src/server/graphql/QueryRootType.js [hunk 2]
```diff
@@ -25,76 +29,93 @@ const QueryRootType = new GraphQLObjectType({
...
+        ? Collections.find('sessionType', {
+          productId: Mongodb.ObjectId(productId)
+        }) : []
+    },
     sessions: {
       type: new GraphQLList(SessionType),
-      args: { sessionTypeId: { type: GraphQLString } },
-      resolve: async (_, { sessionTypeId }) => validateId(sessionTypeId)
+      args: { sessionTypeId: { type: new GraphQLNonNull(GraphQLString) } },
+      resolve: async (root_, { sessionTypeId }) => validateId(sessionTypeId)
```
Comment (0xchimz) [inline src/server/graphql/QueryRootType.js:71]: `root_` -> `root`

File: src/server/graphql/QueryRootType.js [hunk 3]
```diff
@@ -25,76 +29,93 @@ const QueryRootType = new GraphQLObjectType({
...
+      args: { productId: { type: new GraphQLNonNull(GraphQLString) } },
+      resolve: async (root, { productId }) => validateId(productId)
         ? Collections.find('feature', {
           productId: Mongodb.ObjectId(productId)
         }, { _id: -1 }) : []
     },
+    sessionTypes: {
+      type: new GraphQLList(SessionTypeType),
+      args: { productId: { type: new GraphQLNonNull(GraphQLString) } },
+      resolve: async (root_, { productId }) => validateId(productId)
```
Comment (0xchimz) [inline src/server/graphql/QueryRootType.js:63]: `root_` -> `root`

File: src/server/graphql/ResolveHelper.js [hunk 4]
```diff
@@ -32,24 +42,25 @@ export const mapActionsToLinks = session => {
...
+    ...session,
+    actions: removeFocusAndBlur(session.actions)
+  })),
+  paths => _.groupBy(paths, path => (
+    _.reduce(path.actions, (prev, action) => prev + action.actionTypeId, '')
+  )),
+  paths => _.map(Object.keys(paths), pathKey => ({
+    ...paths[pathKey][0],
+    count: paths[pathKey].length,
+  })),
```
Comment (0xchimz) [inline src/server/graphql/ResolveHelper.js:65]:
```javascript
paths => _.map(paths, path => ({ ...paths[0], count: path.length })
```

File: src/shared/nt-results/ResultsPage.react.js [hunk 5]
```diff
@@ -1,8 +1,68 @@
...
 
 import ResultsByLoad from './ResultsByLoad.react'
+import styles from './ResultsPage.styl'
+
+class ResultsPage extends React.Component {
+  static propTypes = {
+    match: PropTypes.shape({
+      url: PropTypes.string.isRequired,
+      params: PropTypes.object.isRequired,
+      isExact: PropTypes.bool.isRequired,
```
Comment (0xchimz) [inline src/shared/nt-results/ResultsPage.react.js:16]: `params` and `isExact` are never used

File: src/shared/nt-results/ResultsPageContainer.react.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    options: ({ productId }) => ({ variables: { productId } })
+  }),
+)
+
+class ResultsPageContainer extends React.Component {
+  static propTypes = {
+    match: PropTypes.shape({
+      url: PropTypes.string.isRequired,
+      params: PropTypes.object.isRequired,
+      isExact: PropTypes.bool.isRequired,
```
Comment (0xchimz) [inline src/shared/nt-results/ResultsPageContainer.react.js:31]: `params` and `isExact` are never used

**Review by 0xchimz — APPROVED** [review]

**Comment by pNagi** [comment]
Note that every comment suggest will be fixed in next branch.

**Pull request merged**